test/integration/query-interface/schemas.test.ts

```diff
@@ -36,4 +39,48 @@ describe('QueryInterface#createSchema', async () => {
 
     expect(queryGeneratorSpy.args[0]).to.include(options);
   });
+
+  it('drops a schema', async () => {
+    await queryInterface.createSchema(testSchema);
+    const preDeletionSchemas = await queryInterface.showAllSchemas();
+    expect(preDeletionSchemas).to.include(testSchema, 'createSchema did not create testSchema');
+
+    await queryInterface.dropSchema(testSchema);
+    const postDeletionSchemas = await queryInterface.showAllSchemas();
+    expect(postDeletionSchemas).to.not.include(testSchema, 'dropSchema did not drop testSchema');
+  });
+
+  it('shows all schemas', async () => {
+    await queryInterface.createSchema(testSchema);
+    const allSchemas = await queryInterface.showAllSchemas();
+
+    const expected = dialectsWithWeirdSchemas.includes(dialect.name)
+      ? [testSchema]
+      : [sequelize.config.database, testSchema];
+    expect(allSchemas.sort()).to.deep.eq(expected.sort());
+  });
+
+  describe('drops all schemas', () => {
+    it('drops all schemas except test schema', async () => {
+      await queryInterface.dropAllSchemas({
+        skip: [sequelize.config.database],
+      });
+      const postDeleteSchemas = await queryInterface.showAllSchemas();
+
+      const expected = dialectsWithWeirdSchemas.includes(dialect.name) ? [] : [sequelize.config.database];
+      expect(postDeleteSchemas).to.deep.eq(expected);
+    });
+
+    it('drops all schemas', async () => {
+      await queryInterface.dropAllSchemas();
+      const postDeleteSchemas = await queryInterface.showAllSchemas();
+      expect(postDeleteSchemas).to.be.empty;
+
+      // Recreate test schema - can't run this in an `after` block since `afterEach` runs first
+      if (!dialectsWithWeirdSchemas.includes(dialect.name)) {
```

Comment:
See my comment above about the name of `dialectsWithWeirdSchemas`; I think we should make sure that only 'mariadb' and 'mysql' run this and not other new dialects